### ISSUE TITLE
UUIDs not being serialized as expected in JAX-RS

### ISSUE THREAD

**Issue by tofflos** [payara/Payara#2541]
# Description #
----------

When returning an entity containing an UUID the UUID is serialized like this:

```
{
    "id": {
        "leastSignificantBits": -7537710357794071555,
        "mostSignificantBits": -5541175049454662535
    },
    "name": "Item A"
} 
```

## Expected Outcome

I expect the UUID to be serialized like this:

```
{
    "id": "b319caa6-a2ff-4c79-9764-ac5db7e6dbfd",
    "name": "Item A"
}
```

## Steps to reproduce (Only for bug reports) 

1 -**  Create a simple JAX-RS resource

```
@Path("/items")
@RequestScoped
@Transactional(Transactional.TxType.REQUIRED)
public class ItemResource {

    private static final List<Item> ITEMS = Arrays.asList(
            new Item(UUID.fromString("b319caa6-a2ff-4c79-9764-ac5db7e6dbfd"), "Item A"),
            new Item(UUID.fromString("78c3492f-f8d2-48cd-9272-3859bd689add"), "Item B"));

    public ItemResource() {
        // Required by RESTEasy
    }

    @GET
    @Produces(MediaType.APPLICATION_JSON)
    public Response doGet() throws JsonProcessingException {
        System.out.println(new ObjectMapper().writeValueAsString(ITEMS)); // Expected
        return Response.ok().entity(ITEMS).build(); // Actual
    }

    @GET
    @Path("{id}")
    @Produces(MediaType.APPLICATION_JSON)
    public Response doGetById(@PathParam("id") UUID id) throws JsonProcessingException {
        Item item = ITEMS.stream().filter(i -> id.equals(i.getId())).findFirst().orElseThrow(() -> new NotFoundException());
        System.out.println(new ObjectMapper().writeValueAsString(item)); // Expected
        return Response.ok(item).build(); // Actual
    }
}
```
2 -**  Deploy the project in Payara Micro:

`java -jar .\payara-micro-5.181.jar --deploy .\target\mavenproject72-1.0-SNAPSHOT.war`

3 -**  Request all the items from the items resource:

```
$ http http://localhost:8080/mavenproject72-1.0-SNAPSHOT/api/v1/items
HTTP/1.1 200 OK
Content-Length: 224
Content-Type: application/json
Server: Payara Micro #badassfish
X-Frame-Options: SAMEORIGIN

[
    {
        "id": {
            "leastSignificantBits": -7537710357794071555,
            "mostSignificantBits": -5541175049454662535
        },
        "name": "Item A"
    },
    {
        "id": {
            "leastSignificantBits": -7894185238746064163,
            "mostSignificantBits": 8701879375396751565
        },
        "name": "Item B"
    }
]
```

Notice my expected behavior in the server log:

```
[
  {
    "id": "b319caa6-a2ff-4c79-9764-ac5db7e6dbfd",
    "name": "Item A"
  },
  {
    "id": "78c3492f-f8d2-48cd-9272-3859bd689add",
    "name": "Item B"
  }
]
```

4 -**  Request a single item from the items resource:

```
$ http http://localhost:8080/mavenproject72-1.0-SNAPSHOT/api/v1/items/b319caa6-a2ff-4c79-9764-ac5db7e6dbfd
HTTP/1.1 200 OK
Content-Length: 111
Content-Type: application/json
Server: Payara Micro #badassfish
X-Frame-Options: SAMEORIGIN

{
    "id": {
        "leastSignificantBits": -7537710357794071555,
        "mostSignificantBits": -5541175049454662535
    },
    "name": "Item A"
}
```

Notice my expected behavior in the server log:

```
{
  "id": "b319caa6-a2ff-4c79-9764-ac5db7e6dbfd",
  "name": "Item A"
}
```

## Environment ##

- **Payara Version**: 5.181
- **Edition**: Micro
- **JDK Version**: 8 u161 - Oracle/OpenJDK
- **Operating System**: Windows

**Comment by tofflos** [comment]
Minimal project that reproduces the error:
[mavenproject72.zip](https://github.com/payara/Payara/files/1822345/mavenproject72.zip)

**Comment by mulderbaba** [comment]
@tofflos Since you're using 5.181, you are depending on [``Yasson``](https://github.com/eclipse-ee4j/yasson) under the hood in order to handle serializing/deserializing of JSON. And with the Yasson, ``UUIDTypeSerializer`` and ``UUIDTypeDeserializer``are not integrated out-of-the-box with its release 1.0.1, which is the current latest available version of the framework and it's the one bundled w/ Payara 5.181 atm. In order to enable this you need to get this [PR](https://github.com/eclipse-ee4j/yasson/pull/80) merged onto the Yasson that you are using and replace the one that resides under glassfish/modules.

**Comment by tofflos** [comment]
I eventually figured out that I could customize the serialization of UUIDs without waiting for a new release of Yasson.

Create the serializer:

```
public class UUIDJsonbSerializer implements JsonbSerializer<UUID> {

    @Override
    public void serialize(UUID obj, JsonGenerator generator, SerializationContext ctx) {
        generator.write(obj.toString());
    }
}
```
Register the serializer:
```
@Provider
public class JsonbContextResolver implements ContextResolver<Jsonb> {
    
    private final Jsonb jsonb;

    public JsonbContextResolver() {
        jsonb = JsonbBuilder.create(new JsonbConfig().withSerializers(new UUIDTypeSerializer()));
    }

    @Override
    public Jsonb getContext(Class<?> type) {
        return jsonb;
    }
}
```
Request all the items from the items resource:

```
$ http http://localhost:8080/mavenproject72-1.0-SNAPSHOT/api/v1/items
HTTP/1.1 200 OK
Content-Length: 125
Content-Type: application/json
Server: Payara Micro #badassfish
X-Frame-Options: SAMEORIGIN

[
    {
        "id": "b319caa6-a2ff-4c79-9764-ac5db7e6dbfd",
        "name": "Item A"
    },
    {
        "id": "78c3492f-f8d2-48cd-9272-3859bd689add",
        "name": "Item B"
    }
]
```

 Request a single item from the items resource:

```
$ http http://localhost:8080/mavenproject72-1.0-SNAPSHOT/api/v1/items/b319caa6-a2ff-4c79-9764-ac5db7e6dbfd
HTTP/1.1 200 OK
Content-Length: 61
Content-Type: application/json
Server: Payara Micro #badassfish
X-Frame-Options: SAMEORIGIN

{
    "id": "b319caa6-a2ff-4c79-9764-ac5db7e6dbfd",
    "name": "Item A"
}

```

**Comment by rlfnb** [comment]
@mulderbaba It might be already fixed as Yasson 1.0.2 is released and already added to the pom.xml in payara?

**Comment by rdebusscher** [comment]
Newer version of Yasson is integrated and UUID serialization is correct in Payara 5.194.
